code/modules/mob/living/simple_animal/hostile/wendigo.dm

```diff
@@ -0,0 +1,221 @@
+	Can mimic voices, steal poly code, and change it (Maybe just keep a log of people consumed,
+	and use their name plus shout for help)
+	Butcherable for one piece of meat only. When consumed (Don't use reagents) will begin to turn the person
+*/
+#define HUMEVOLV 1
+#define EVOLEVOLV 3
+
+
+/mob/living/simple_animal/hostile/wendigo
+	name = "wendigo"
+	desc = "Standing tall, deep sunken eyes, and a voracious appetite. This individual has seen better days."
+	icon_state = "wendigo"
+	icon_living = "wendigo"
+	icon_dead = "wendigo_dead"
+
+	health = 150
+	maxHealth = 150
+	speed = 1
+	stat_attack = DEAD //Gotta chow down somehow
+	vision_range = 12 //Slightly larger vision range
+	harm_intent_damage = 8
+	melee_damage_lower = 15
+	melee_damage_upper = 20
+	attacktext = "bites"
+	attack_sound = 'sound/weapons/bite.ogg'
+	meat_amount = 1 //Only one piece of meat
+	meat_type = /obj/item/weapon/reagent_containers/food/snacks/meat/wendigo
+	var/list/names = list() //List of names of the people it's eaten
+	var/consumes = 0 //How many people it has eaten
+	speak_chance = 15
+	speak = list("Help!","Help me!","Somebody help!","Get over here, quickly!")
+
+
+/mob/living/simple_animal/hostile/wendigo/CanAttack(var/atom/the_target)
+	var/list/target_prox = view(the_target, vision_range)
+	for(var/obj/machinery/space_heater/campfire/fire in target_prox)
+		var/dist = get_dist(the_target, fire)
+		if(dist < (fire.light_range*2))//Just sitting on the edge of the fire
+			return
+
+	if(isliving(target) && !ishuman(target))
+		var/mob/living/L = target
+		if(L.isDead())
+			return
+
+	return ..(the_target)
+
+/mob/living/simple_animal/hostile/wendigo/AttackingTarget()
+	if(!target)
+		return
+
+	if(ismob(target))
+		var/mob/living/mob_target = target
+		if(mob_target.isDead() && !istype(mob_target, /mob/dead/observer) && (health < maxHealth || ishuman(mob_target)))
+			set waitfor = 0
+			visible_message("<span class = 'notice'>\The [src] starts to take a bite out of \the [target].</span>")
+			stop_automated_movement = 1
+			if(do_after(src, mob_target, 50, needhand = FALSE))
+				playsound(src, 'sound/weapons/bite.ogg', 50, 1)
+				var/damage = rand(melee_damage_lower, melee_damage_upper)
+				mob_target.adjustBruteLoss(damage)
+				if(health < maxHealth)
+					health = min(maxHealth,(health+damage))
+
+				if(ishuman(mob_target))
+					if(mob_target.health < -400)
+						visible_message("<span class = 'warning'>\The [src] is trying to consume \the [mob_target]!</span>","<span class = 'warning'>You hear crunching.</span>")
+						if(do_after(src, mob_target, 50, needhand = FALSE))
+							consumes += 1
+							if(!names.Find(mob_target.real_name))
+								names.Add(mob_target.real_name)
+							mob_target.gib()
+
+			return
+	. =..()
+
+/mob/living/simple_animal/hostile/wendigo/Life()
+	/*Check evolve like zombies, run away from fire,
+	mimic speech if anyone's nearby but just out of eyesight
+	*/
+	..()
+	if(!isDead())
+		check_evolve()
```

Comment:
this proc may `qdel(src)`, surely we don't want to do anything else in here if that's the case
I suggest you make it return a value and check it